### PR TITLE
Add integration testing framework and tests

### PR DESCRIPTION
**Pull request by a-hilaly**
This PR adds a comprehensive integration testing framework for Symphony along
with several bug fixes and improvements that been discovered while writing the
tests.

#### Integration Testing Framework
- Set up integration test environment using `envtest` and `ginkgo/gomega`
- Add test suites covering core controller functionality:
  - CRD management and validation
  - Resource topology handling 
  - Graph validation
  - Status management
- Include integrations workflow tests for:
  - `NetworkingStack` (vpc, subnets, security groups)
  - `DeploymentService` RG
  - `EKSCluster` provisioning
- Add ACK CRDs (`ec2`, `eks`, `iam`) required for testing AWS resource workflows

#### Bug Fixes & Improvements
- Make `StartServingGVK` idempotent to handle duplicate GVK registrations gracefully
- Fix type assertion in `gracefulShutdown` to properly handle informer wrapper
- Correct status variable path construction by adding proper dot notation
- Use correct metadata package (`metav1` instead of `corev1`) for `ConditionStatus`

#### Testing:
The new testing framework provides coverage for:
- Core functionality: CRD manipulation, validation, topology, status management
- Complex workflows: Multi resource orchestration with dependencies
- Edge cases: Resource naming, validation failures, circular dependencies

#### How to Run Tests
```bash
# Run all integration tests
go test ./test/integration/... -v

# Run specific test suite
go test ./test/integration/suites/core -v
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/README.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+informers, and client-go. The testing strategy aims to ensure that Symphony works as
+expected in a Kubernetes environment, and that it can be safely deployed in production
+clusters.
+
+## Technical principles for testing
+
+1. Use existing Kubernetes testing frameworks (when possible): Don't reinvent the wheel. 
+    If a feature is not covered by existing frameworks, contribute to them.
+3. Focus on Symphony's logic, not on other controllers or Kubernetes components. e.g
+   avoid testing antive controllers, ACK or Karpenter's behaviour...
```
Comment (michaelhtm) [inline test/README.md]: nit:
```suggestion
   avoid testing native controllers, ACK or Karpenter's behaviour...
```

File: test/integration/crds/ack-ec2-controller/ec2.services.k8s.aws_elasticipaddresses.yaml [hunk 2]
```diff
@@ -0,0 +1,182 @@
+---
```
Comment (michaelhtm) [inline test/integration/crds/ack-ec2-controller/ec2.services.k8s.aws_elasticipaddresses.yaml:1]: Why not use the CRD resource group here? with conditionals you can specify which CRD you want to deploy/manage

Comment (a-hilaly) [reply test/integration/crds/ack-ec2-controller/ec2.services.k8s.aws_elasticipaddresses.yaml:1]: This is such a great idea Michael! Here is my thought here, we need to tweak the environement setup, to support installing resourcegroups instead of CRDs, for now they only know how to install CRDs.
